update branch policy (#407)
nautobot/nautobot-plugin-golden-config: docs/dev/dev_contributing.md

The branching policy includes the following tenets:

> - The develop branch is the branch for bug fixes.

Reviewer: ```suggestion
- The `develop` branch is the branch for bug fixes.
```

> - The next branch is the branch the next major and minor paired version planned.

Reviewer: ```suggestion
- The next branch is the branch for new features. It is therefore the branch from which the next planned `stable-<major>.<minor>` version will be sourced.
```

Author: I don't read this as accurate `It is therefore the branch from which the next planned `stable-<major>.<minor>` version will be sourced.`

- The `stable-<major>.<minor>` branch is the branch of the latest version within that major/minor version.
- The `stable-<major>.<minor>` branch will have all of the latest bug fixes and security patches, and may or may not represent the released version.

> - PRs intended to add new features should be sourced from the next branch.

Reviewer: ```suggestion
- PRs intended to add new features should be branched from and merged to the `next` branch.
```

When a new release of any kind (e.g. from `develop` to `main`, or a release of a `stable-<major>.<minor>`) is created the following should happen.

> - A release PR is created with:

Reviewer: ```suggestion
- A release PR is created:
```

    - The title should be in the form of `v<major>.<minor>.<patch>`.
    - The description should be the changes that were added to the `version_<major>.<minor>.md` document.

> - If merged into `main`, then push from `main` to `develop`, in order to retain the merge commit created when the PR was merged

Reviewer: ```suggestion
- If merged into `main`, then push from `main` to `develop`, in order to retain the merge commit created when the PR was merged.
```

- If the is a new `<major>.<minor>`, create a `stable-<major>.<minor>` for the **previous** version, so that security updates to old versions may be applied more easily.

> - A post release PR is created with.

Reviewer: ```suggestion
- A post release PR is created:
```
